### DISCUSSION TITLE
Upgrade - how to update old templates to the newest version?

### DISCUSSION THREAD

**Discussion by acip** [GrapesJS/grapesjs#3633] [Q&A]
I have some templates  that were generated with v0.14.5. How can I "upgrade" them to the latest version? They have classes attached and I see that the newest version doesn't add classes to the elements anymore.

Context: we use bespoke components and some defaults (e.g. draggable, resizable etc., style) have been changed.

If more details are required, I can provide them.

Thank you for this great project!

**Comment by artf** [comment]
> They have classes attached and I see that the newest version doesn't add classes to the elements anymore.

I'm not sure what you're talking about, classes should be imported as before.

> Context: we use bespoke components and some defaults (e.g. draggable, resizable etc., style) have been changed.

If your custom components were relying on some previous default property you would just need to update your components but honestly, I don't even remember such a change on any core component.

Let me see some template code not loading properly, please.

**Comment by acip** [comment]
Hey, thanks for the quick response. Here is the template generated by v0.14.5 https://jsonblob.com/535ae06f-d4f6-11eb-8bc1-415e99da64cb

This is how it looks https://ibb.co/Zz8qg2J


We've changed some custom components' default attributes and style and changed the dragMode='absolute'. One of the issues is that the old components can't be moved around. 

I can provide code samples if needed.

**Comment by acip** [comment]
I've setup old vs new version for an existing template:

https://grapesjs-templates-upgrade.ciprianamariei.repl.co/0.14.5-old-template.html
https://grapesjs-templates-upgrade.ciprianamariei.repl.co/0.16.45-old-template.html

![oldvsnew](https://user-images.githubusercontent.com/1649442/123832900-80497780-d90e-11eb-9dc0-f2ecb4f14e15.png)

**Reply by artf** [reply]
Well, you have actually an error in the editor, this is why is not responding. 
Seems like there is an issue with block container rendering. 
Is there any reason for using 0.16.x instead of 0.17.x? We don't actually support/fix previous versions.
